src/OrleansAWSUtils/Providers/Storage/DynamoDBStorage.cs

```diff
@@ -0,0 +1,415 @@
+
+        private bool useJsonFormat;
+        private JsonSerializerSettings jsonSettings;
+
+        private GrainStateTableDataManager tableDataManager;
+
+        /// <summary> Name of this storage provider instance. </summary>
+        /// <see cref="IProvider.Name"/>
+        public string Name { get; private set; }
+
+        /// <summary> Logger used by this storage provider instance. </summary>
+        /// <see cref="IStorageProvider.Log"/>
+        public Logger Log { get; private set; }
+
+        /// <summary> Default constructor </summary>
+        public DynamoDBStorage()
+        {
+            tableName = TableNameDefaultValue;
+            id = Interlocked.Increment(ref counter);
+        }
+
+        /// <summary> Initialization function for this storage provider. </summary>
+        /// <see cref="IProvider.Init"/>
+        public Task Init(string name, IProviderRuntime providerRuntime, IProviderConfiguration config)
+        {
+            Name = name;
+            serviceId = providerRuntime.ServiceId.ToString();
+
+            if (config.Properties.ContainsKey(ServicePropertyName))
+                service = config.Properties[ServicePropertyName];
+
+            if (config.Properties.ContainsKey(AccessKeyPropertyName))
+                accessKey = config.Properties[AccessKeyPropertyName];
+
+            if (config.Properties.ContainsKey(SecretKeyPropertyName))
+                secretKey = config.Properties[SecretKeyPropertyName];
+
+            if (config.Properties.ContainsKey(ReadCapacityUnitsPropertyName))
+                readCapacityUnits = int.Parse(config.Properties[ReadCapacityUnitsPropertyName]);
+
+            if (config.Properties.ContainsKey(WriteCapacityUnitsPropertyName))
+                writeCapacityUnits = int.Parse(config.Properties[WriteCapacityUnitsPropertyName]);
+
+            if (config.Properties.ContainsKey(TableNamePropertyName))
+                tableName = config.Properties[TableNamePropertyName];
+
+            isDeleteStateOnClear = config.Properties.ContainsKey(DeleteOnClearPropertyName) &&
+                "true".Equals(config.Properties[DeleteOnClearPropertyName], StringComparison.OrdinalIgnoreCase);
+
+            Log = providerRuntime.GetLogger("Storage.AWSDynamoDBStorage." + id);
+
+            var initMsg = string.Format("Init: Name={0} ServiceId={1} Table={2} DeleteStateOnClear={3}",
+                Name, serviceId, tableName, isDeleteStateOnClear);
+
+            if (config.Properties.ContainsKey(UseJsonFormatPropertyName))
+                useJsonFormat = "true".Equals(config.Properties[UseJsonFormatPropertyName], StringComparison.OrdinalIgnoreCase);
+
+            this.jsonSettings = SerializationManager.UpdateSerializerSettings(SerializationManager.GetDefaultJsonSerializerSettings(), config);
+
+            initMsg = String.Format("{0} UseJsonFormat={1}", initMsg, useJsonFormat);
+
+            Log.Info(ErrorCode.StorageProviderBase, "AWS DynamoDB Provider: {0}", initMsg);
```

Comment:
I suggest you either use the Azure error codes (like in the previous version of this PR) or copy paste and rename them like we did in the ado.net storage provider PR. If you feel like fixing this the right way, the best would be to merge all the storage provider errors and put them in the Orleans providers project. see https://github.com/dotnet/orleans/pull/1682#issuecomment-236450895. the important thing is that you shouldn't use the base error because it'd make it harder later to refactor.